packages/components/src/scroll-to-top/scroll-to-top-on-mount.tsx

```diff
@@ -0,0 +1,10 @@
+import { useEffect } from 'react';
+
+export function ScrollToTopOnMount() {
+	useEffect( () => {
+		if ( ! window.location.hash ) {
```

Comment:
Can we avoid scrolling if the user simply refreshes the page (and expects their scroll position to persist)?

Reply:
> Because `<Main>` is used on all views, this seems like it has the potential to accidentally break lots of things.
> 
> What's the best way to guard against this, and roll the change out safely?

We can deploy it behind a feature flag on `stage` and deploy it to production after ~2 weeks.